pkg/segment/structs/segstructs.go

```diff
@@ -259,18 +260,27 @@ type QueryCount struct {
 // A helper struct to keep track of errors and results together
 // In cases of partial failures, both logLines and errList can be defined
 type NodeResult struct {
-	AllRecords       []*utils.RecordResultContainer
-	ErrList          []error
-	Histogram        map[string]*AggregationResult
-	TotalResults     *QueryCount
-	RenameColumns    map[string]string
-	SegEncToKey      map[uint16]string
-	TotalRRCCount    uint64
-	MeasureFunctions []string        `json:"measureFunctions,omitempty"`
-	MeasureResults   []*BucketHolder `json:"measure,omitempty"`
-	GroupByCols      []string        `json:"groupByCols,omitempty"`
-	Qtype            string          `json:"qtype,omitempty"`
-	BucketCount      int             `json:"bucketCount,omitempty"`
+	AllRecords                    []*utils.RecordResultContainer
+	ErrList                       []error
+	Histogram                     map[string]*AggregationResult
+	TotalResults                  *QueryCount
+	RenameColumns                 map[string]string
+	SegEncToKey                   map[uint16]string
+	TotalRRCCount                 uint64
+	MeasureFunctions              []string          `json:"measureFunctions,omitempty"`
+	MeasureResults                []*BucketHolder   `json:"measure,omitempty"`
+	GroupByCols                   []string          `json:"groupByCols,omitempty"`
+	Qtype                         string            `json:"qtype,omitempty"`
+	BucketCount                   int               `json:"bucketCount,omitempty"`
+	PerformAggsOnRecs             bool              // if true, perform aggregations on records that are returned from rrcreader.go
+	RecsAggsType                  []PipeCommandType // To determine Whether it is GroupByType or MeasureAggsType
+	GroupByRequest                *GroupByRequest
+	MeasureOperations             []*MeasureAggregator
+	RecsAggsRecords               map[string]map[string]interface{}
+	RecsAggsRunningBuckets        interface{} // Evaluates to []*blockresults.RunningBucketResults
+	RecsAggsStringBucketIdx       map[string]int
+	RecsAggsProcessedSegments     uint64
+	RecsAggsProcessedSegmentsLock sync.Mutex
```

Comment:
For these new fields, it would be good if you could group them into a new struct type (or a few new types, if that makes more sense) just so the `NodeResult` doesn't get too bloated and do make it a little easier to understand the grouping of the fields (e.g., I assume the `RecsAggs...` fields are only used for certain queries, and for those queries all of those attributes will be needed, so you could make a new struct type for those)

Reply:
I have removed some of the types in the struct that I am not using.
Let me know if you still want me to reduce then I will change it accordingly.